developersociety/django-template
# Fix SVG Spritemap plugin and add SVG icon template

Sources
https://app.forecast.it/project/P-57/scoping/T6539

Description
- The SVG spritemap generation wasn't working, so this fixes it
- Also creates a template for icons that makes it a bit simpler to include them (slightly less code)
- Adds svg4everybody which lets SVG spritemaps work in IE11
- Adds the spritemap webpack config to the production webpack config so it'll work in production too!

Setup instructions
- Pull down this branch
- Install cookiecutter with `pip install --user cookiecutter` if you don't have it already 
- Deactivate the virtual environment and navigate to the Sites directory
- Use cookiecutter to create a new project from this template using `cookiecutter ~/Sites/django-template` - if this doesn't work and it says cookiecutter isn't installed, try `python -m cookiecutter ~/Sites/django-template/`
- Navigate into that new project and create a virtual environment for it with `mkvirtualenv yourprojectname`
- `make reset`

How to test
- Run your test project with `./manage.py runserver`
- Go to http://127.0.0.1:8000/ and check that the icons in the footer are shown, and if you resize to mobile that the burger/close icon for the menu work
- Add a new icon in the src/sprite/svg folder - I copied the 'alert' icon as it'll then appear at the start of the generated file
- View the /dist/svg/sprite.svg file and check that the icon you copied has appeared in the file, and that if you delete it it's automatically removed again
- Load up IE11 in Browserstack and check that the footer icons are visible

Happy to hear any suggestions on how this could be made better, and if you think the icon template is useful!


For comprehensive guidelines on code review at DEV, see here: https://www.notion.so/developersociety/Code-Review-b6d646b1d6e54011ada8169aee543231.
